compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/DesktopComposeSceneLayer.desktop.kt

```diff
@@ -45,12 +120,72 @@ internal abstract class DesktopComposeSceneLayer : ComposeSceneLayer {
     }
 
     /**
-     * Renders the overlay on the main Compose view canvas.
+     * Called when the layers in [composeContainer] have changed.
+     */
+    open fun onLayersChange() {
+    }
+
+    /**
+     * Renders an overlay on the canvas.
+     *
+     * @param canvas the canvas to render on
+     * @param width the width of the overlay
+     * @param height the height of the overlay
+     * @param transparent a flag indicating whether [canvas] is transparent
+     */
+    open fun onRenderOverlay(canvas: Canvas, width: Int, height: Int, transparent: Boolean) {
+    }
+
+    /**
+     * This method is called when a mouse event occurs outside of this layer.
      *
-     * @param canvas the canvas of the main Compose view
-     * @param width the width of the canvas
-     * @param height the height of the canvas
+     * @param event the mouse event
+     */
+    fun onMouseEventOutside(event: MouseEvent) {
+        if (isClosed || !event.isMainAction() || inBounds(event)) {
+            return
+        }
+        val eventType = when (event.id) {
+            MouseEvent.MOUSE_PRESSED -> PointerEventType.Press
+            MouseEvent.MOUSE_RELEASED -> PointerEventType.Release
+            else -> return
+        }
+        outsidePointerCallback?.invoke(eventType)
+    }
+
+    private fun inBounds(event: MouseEvent): Boolean {
+        val point = if (event.component != windowContainer) {
+            SwingUtilities.convertPoint(event.component, event.point, windowContainer)
+        } else {
+            event.point
+        }
+        return boundsInWindow.toAwtRectangle(density).contains(point)
+    }
+
+    /**
+     * Detect and trigger [DesktopComposeSceneLayer.onMouseEventOutside] if event happened below
+     * focused layer.
      */
-    open fun onRenderOverlay(canvas: Canvas, width: Int, height: Int) {
+    private inner class DetectEventOutsideLayer : AwtEventFilter {
+        override fun shouldSendMouseEvent(event: MouseEvent): Boolean {
+            layersAbove.toList().fastForEachReversed {
+                it.onMouseEventOutside(event)
```

Comment:
Better to call it somewhere else, not inside `shouldSendMouseEvent`, that by its name shouldn't generate side effects.

Or rename `AwtEventFilter` to `AwtEventHandler`, `shouldSendMouseEvent` to `handleSendMouseEvent`

Reply:
Yep, thought about it, but it caused more unrelated changes